client/test/lib/helpers/myaccounthelpers.coffee

```diff
@@ -40,30 +47,59 @@ module.exports =
       .click                   saveButtonSelector
       .waitForElementVisible   '.kdnotification.main', 20000
       .refresh()
+      .pause 2000
       .waitForElementVisible   lastnameSelector, 20000
       .assert.value            lastnameSelector, newLastName
       .pause  1000, callback
 
-  #UpdateEmail will be reimplemented
-  # updateEmail: (browser, callback) ->
-  #   newEmail           = newName + newLastName + '@koding.com'
-  #   browser
-      # .waitForElementVisible   emailSelector, 20000
-      # .clearValue              emailSelector
-      # .setValue                emailSelector, newName + '\n'
-      # .click                   saveButtonSelector
-      # .waitForElementVisible   '.kdmodal.kddraggable', 20000
-      # .pause 3000
-      # .waitForElementVisible   emailSelector, 20000
-      # .getValue                emailSelector, (result) ->
-      #   assert.equal           result.value, newName
+
+  updateEmailWithInvalidPassword: (browser, callback) ->
+    newEmail = 'wrongemail@koding.com'
+    browser
+      .refresh()
+      .pause 2000
+      .waitForElementVisible   emailSelector, 20000
+      .clearValue              emailSelector
+      .setValue                emailSelector, newEmail + '\n'
+      .click                   saveButtonSelector
+      .waitForElementVisible   modalSelector, 20000
+      .assert.containsText     '.ContentModal.content-modal header > h1', 'Please verify your current password'
+      .setValue                passwordSelector, '123456'
+      .click                   confirmEmailButton
+      .waitForElementVisible   notificationSelector, 20000
+      .assert.containsText     notificationSelector, 'Current password cannot be confirmed'
+      .pause 1000, callback
+
+
+  updateEmailWithInvalidPin: (browser, callback) ->
+    newEmail = 'wrongemail2@koding.com'
+    browser
+      .refresh()
+      .pause 3000
+      .waitForElementVisible   emailSelector, 20000
+      .clearValue              emailSelector
+      .setValue                emailSelector, newEmail + '\n'
+      .click                   saveButtonSelector
+      .pause 2000
+      .waitForElementVisible   modalSelector, 20000
+      .waitForElementVisible   passwordSelector, 30000
+      .setValue                passwordSelector, user.password
+      .click                   confirmEmailButton
+      .waitForElementVisible   modalSelector, 20000
+      .waitForElementVisible   pinSelector, 2000
+      .setValue                pinSelector, '1234'
+      .click                   updateEmailButton
+      .waitForElementVisible   notificationSelector, 20000
+      .assert.containsText     notificationSelector, 'PIN is not confirmed.'
+      .pause  1, callback
 
 
   updatePassword: (browser, callback) ->
-    user            =  utils.getUser()
     currentPassword = user.password
     newPassword     = utils.getPassword()
     browser
+      .refresh()
+      .pause 3000
```

Comment:
it is better we use .waitForElementVisible function here instead of waiting 3 seconds. It may take longer sometimes.